File: exercise/C#/day08/ToyProduction/Domain/Toy.cs
namespace ToyProduction.Domain
{
    public class Toy(string name, State state)
    {
        public string Name { get; } = name;
        public State State { get; set; } = state;
    }

    public enum State
    {
        Unassigned,
        InProduction,
        Completed
    }
}

File: solution/day08/src/main/java/domain/Toy.java
package domain;

public class Toy {
    public enum State {
        UNASSIGNED, IN_PRODUCTION, COMPLETED
    }

    private final String name;
    private State state;

    public Toy(String name) {
        this.name = name;
        this.state = State.UNASSIGNED;
    }

    public String getName() {
        return name;
    }

    public State getState() {
        return state;
    }

    public void assignToElf() {
        if(state == State.UNASSIGNED)
            this.state = State.IN_PRODUCTION;
    }
}